File: Python/bbsort.py
'''
冒泡排序
Python 实例
By WeepingDogel
'''
def sort(arr):
    for i in range(1,len(arr)):
        for j in range(0, len(arr) - i):
            if arr[j] > arr[j+1]:
                arr[j],arr[j+1] = arr[j+1],arr[j]
    return arr
arr = [73837372882, 2772828191919, 82828282828,828282728181]

print(sort(arr))

File: C/bbsort.c
/*
冒泡排序 
C 语言实例
By WeepingDogel
*/
#include<stdio.h>

void sort(int arr[],int len)
{
    int i,j,temp;
    for(i = 0; i < len - 1; i++)
    {
        for(j = 0; j < len - 1 - i;j++)
        {
            if (arr[j] > arr[j + 1])
            {
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j + 1] =temp;
            }
        }
    }
}
int main()
{
    int n1,n2,n3,n4;
    printf("type four numbers:");
    scanf("%d%d%d%d",&n1,&n2,&n3,&n4);
    int arr[] = {n1,n2,n3,n4};
    int len;
    len = (int)sizeof(arr) / sizeof(*arr);
    sort(arr,len);
    int i;
    for(i = 0; i < len; i++)
    {
        printf("%d ",arr[i]);
    }
    return 0;
}